src/btree/bt_sync_obsolete.c

```diff
@@ -94,11 +105,41 @@ __sync_obsolete_inmem_evict(WT_SESSION_IMPL *session, WT_REF *ref)
         /* Mark the obsolete page to evict soon. */
         __wt_page_evict_soon(session, ref);
         WT_STAT_CONN_DSRC_INCR(session, checkpoint_cleanup_pages_evict);
+    } else {
+        /*
+         * Limit the number of obsolete time window pages that are marked as dirty to reduce the
+         * load.
+         */
+        if (S2BT(session)->cc_obsolete_tw_pages >=
+          S2C(session)->heuristic_controls.cc_obsolete_tw_pages_dirty)
+            return (0);
+
+        /*
+         * The pages that are completely removed are eliminated once they are obsolete. There is no
+         * point in clearing their on-disk time window information now.
+         */
+        if (WT_TIME_AGGREGATE_HAS_STOP(&newest_ta))
+            return (0);
+
+        newest_ts = WT_MAX(newest_ta.newest_start_durable_ts, newest_ta.newest_stop_durable_ts);
+        if ((newest_ta.newest_txn != WT_TXN_NONE || newest_ts != WT_TS_NONE) &&
```

Comment:
Let's split this `if` as well:
```
/* If there is no transaction or timestamp information available, there is nothing to do. */
if (newest_ta.newest_txn == WT_TXN_NONE && newest_ts == WT_TS_NONE)
  return (0);

/* Ensure the time window information has content that is globally visible. */
if (!__wt_txn_visible_all(session, newest_ta.newest_txn, newest_ts))
  return (0);
```

This code is very similar to the eviction flow, I wonder if we could use the same function or somehow refer them so the code is changed at both locations whenever needed.

Reply:
I prefer to keep it because, in most cases, there is at least one valid value, so with the existing approach, there is no need to check two values before checking the visible all. If we change as suggested, we always need to check both values before deciding.

I added a function to keep the approach the same in three places.